### DISCUSSION TITLE
UMEP URock processor - stacked blocks

### DISCUSSION THREAD

**Discussion by pdugue** [UMEP-dev/UMEP#535] [Q&A] [answered]
Hi,

I am using URock and have a question about the step UMEP Processor URock.
Indeed, I have some buildings in a small area that are close fitted and when I start calculations, the last step never ends : 
![MicrosoftTeams-image (1)](https://github.com/UMEP-dev/UMEP/assets/138692633/29ad6c9a-11d7-4cc8-be38-58cc2dd239ad)
I have identified which building raises the issue : 
![MicrosoftTeams-image](https://github.com/UMEP-dev/UMEP/assets/138692633/7134edb9-f506-4b62-a138-dc2c9d5b6b05)
Without this building, calculations continue and end.
I read this part in the URock documentation, that seems to be linked with my issue : 
![image](https://github.com/UMEP-dev/UMEP/assets/138692633/055e75dd-6878-4ba1-ab88-296c8f573ac5)
Is there any solution to consider this building in calculations ?

Thanks for help ! :) 

Pauline

**Comment by j3r3m1** [comment]
Thank you for reporting. Can you share your building data ?

**Reply by pdugue** [reply]
yes, here is building data : 
[buildings data.zip](https://github.com/UMEP-dev/UMEP/files/12040274/buildings.data.zip)
You will find 2 shapefiles : 
- Batiments_Pb : a small area with the buildings as seen on the picture above and on which I am testing
- Batiments_Petite_Zone_All : the entire area that I want to treat in the study case
Thanks

**Comment by j3r3m1** [comment]
The small zone works for me for a wind coming from west (270°). Both files create an error ? And for which wind direction ?

**Reply by pdugue** [reply]
I have a wind coming from SE (135°)
Yes both files create an error

**Comment by j3r3m1** [comment]
OK thanks the bug happens when cavity and wake zones are created. I am investigating the reason

**Reply by pdugue** [reply]
OK thanks a lot for your help !

**Comment by j3r3m1** [comment, marked as answer]
It should be solved with the last version (https://github.com/UMEP-dev/UMEP-processing/commit/b096e9f0e77bf5b8ddf0e89f133377d9d9ece4a9). Can you have a test downloading the last version as a zip ?

**Reply by pdugue** [reply]
Thanks for your help, I have downloaded both files : "GlobalVariables.py" and "Zones.py". How can I integrate them to QGIS ? Do I have to install again UMEP on QGIS ?

**Reply by j3r3m1** [reply]
Sorry it was not that clear. For each new version of UMEP that has not been released yet (development version) you can download it as a zip file and then install directly the plugin in QGIS using this zip.

To download, you have to go on the UMEP-processing github page and then click "Code" and "download as zip". This is equivalent to this link: https://github.com/UMEP-dev/UMEP-processing/archive/refs/heads/main.zip

Then to install it in QGIS you have to click on "Plugin" -> "Manage and Install Plugins" -> "Install from zip" ad then select the downloaded zip file (more detail here: https://umep-docs.readthedocs.io/en/latest/Getting_Started.html#installing-development-release-could-be-unstable

**Reply by pdugue** [reply]
Hi, 
Thanks for your explainations. I have downloaded it and install in QGIS. The test is a success on the entire study area thanks ! 
Best, 
Pauline

**Comment by 4ngryB1rdGIStudent** [comment]
Hey!

I've encountered the same issue as Pauline but I'm afraid the solution was of no use to me. I've run URock many times and the issue occurs at different wind direction values for different input data. For this particular dataset, the issue occurs at wind directions 185 and 195:
[URock.zip](https://github.com/UMEP-dev/UMEP/files/14510922/URock.zip)

In other instances, the algorithm does indeed produce an error message instead of simply never ending its run (see pictures). With this dataset, this error occurs at wind direction 125:
[URock2.zip](https://github.com/UMEP-dev/UMEP/files/14511022/URock2.zip)
 
![Skärmbild 2024-03-06 151751](https://github.com/UMEP-dev/UMEP/assets/132443956/eec24b67-8c23-4ede-a024-8ee4c09d6892)
![Skärmbild 2024-03-06 151854](https://github.com/UMEP-dev/UMEP/assets/132443956/041c171e-5516-4f4a-9cf0-2de54e0d699e)

Any explanations/suggestions?

Thanks for the help!

Kind regards, Emil
